onnxruntime/core/providers/cuda/shared_inc/cuda_utils.h

```diff
@@ -33,5 +34,25 @@ std::unique_ptr<IConstantBuffer<T>> CreateConstantOnes();
 template <typename T>
 void Fill(T* output, T value, int64_t count);
 
+/*
+  This is a utility wrapper for arbitrary type array
+  Commonly used for passing small list of metadata during cuda kernel launch
```

Comment:
Scott McKay
Would be good to have some documentation saying what TArray should be used for, why and when.

Sherlock Huang
documented.